framework-plugins/lisk-framework-chain-connector-plugin/test/unit/plugin.spec.ts

```diff
@@ -60,33 +79,347 @@ describe('ChainConnectorPlugin', () => {
+		};
+
+		let blsKeyToBFTWeight: Record<string, bigint> = {
+			ad0076aa444f6cda608bb163c3bd77d9bf172f1d2803d53095bc0f277db6bcb3: BigInt(1),
+		};
+
+		beforeEach(async () => {
+			await chainConnectorPlugin.init({
+				logger: testing.mocks.loggerMock,
+				config: {
+					mainchainIPCPath: '~/.lisk/mainchain',
+					sidechainIPCPath: '~/.list/sidechain',
+				},
+				appConfig: testing.fixtures.defaultConfig as never,
+			});
+
+			jest.spyOn(apiClient, 'createIPCClient').mockResolvedValue({} as never);
+			await chainConnectorPlugin.load();
+
+			jest.spyOn(dbApi, 'getChainConnectorInfo').mockResolvedValue({
+				blockHeaders: [
+					{
+						...block.header,
+						height: 1,
+					},
+					{
+						...block.header,
+					},
+				],
+				validatorsHashPreimage: [
+					{
+						validatorsHash: block.header.validatorsHash,
+						validators: [
+							{
+								bftWeight: BigInt(0),
+								blsKey: Buffer.from('00', 'hex'),
+							},
+						],
+					},
+				],
+			} as never);
+		});
+
+		describe('getCertificateFromAggregateCommit', () => {
+			it('should call getChainConnectorInfo', async () => {
+				await chainConnectorPlugin.getCertificateFromAggregateCommit(aggregateCommit);
+
+				expect(dbApi.getChainConnectorInfo).toHaveBeenCalledTimes(1);
+			});
+
+			it('should compute Certificate from BlockHeader', async () => {
+				const blockHeader: chain.BlockHeader = new chain.BlockHeader(block.header);
+				const expectedCertificate = computeCertificateFromBlockHeader(blockHeader);
+				expectedCertificate.aggregationBits = Buffer.alloc(0);
+				expectedCertificate.signature = Buffer.alloc(0);
+
+				const certificate = await chainConnectorPlugin.getCertificateFromAggregateCommit(
+					aggregateCommit,
+				);
+
+				expect(certificate).toEqual(expectedCertificate);
+			});
+		});
+
+		describe('getNextCertificateFromAggregateCommits', () => {
+			let expectedCertificate: Certificate;
+
+			beforeEach(() => {
+				const blockHeader: chain.BlockHeader = new chain.BlockHeader(block.header);
+				expectedCertificate = computeCertificateFromBlockHeader(blockHeader);
+
+				chainConnectorPlugin['_mainchainAPIClient'].invoke = jest
+					.fn()
+					.mockResolvedValue(bftHeights);
+
+				chainConnectorPlugin.checkChainOfTrust = jest
+					.fn()
+					.mockResolvedValueOnce(false)
+					.mockResolvedValueOnce(true);
+				chainConnectorPlugin.getCertificateFromAggregateCommit = jest
+					.fn()
+					.mockResolvedValue(expectedCertificate);
+			});
+
+			it('should call getChainConnectorInfo', async () => {
+				await chainConnectorPlugin.getNextCertificateFromAggregateCommits(2, aggregateCommits);
+
+				expect(dbApi.getChainConnectorInfo).toHaveBeenCalledTimes(1);
+			});
+
+			it('should invoke consensus_getBFTHeights on _mainchainAPIClient', async () => {
+				await chainConnectorPlugin.getNextCertificateFromAggregateCommits(2, aggregateCommits);
+
+				expect(chainConnectorPlugin['_mainchainAPIClient'].invoke).toHaveBeenCalledWith(
```

Comment:
based on the requested change above
```suggestion
				expect(chainConnectorPlugin['_sidechainAPIClient'].invoke).toHaveBeenCalledWith(
```